### ISSUE TITLE
[MD007] Inconsistent indentation for nested lists

### ISSUE THREAD

**Issue by lucaspar** [rvben/rumdl#504]
### rumdl version

0.1.43

### Installation method

Other (`uvx rumdl check`)

### Bug description

Rumdl complains when using nested lists with 4 space indents, expecting 3 spaces:

![Image](https://github.com/user-attachments/assets/fc021e65-3cbd-4580-bc54-eb146c10319a)

### Steps to reproduce

Raw text:

```md
# Things

+ An unordered list
    + An item with 4 spaces, ok.

1. A numbered list
    + A sublist with 4 spaces, not ok
        + A sub item with 4 spaces, ok
    + Why is rumdl expecting 3 spaces for a 4 space indent?
2. Item 2
3. Item 3
```

### Expected behavior

Don't complain with 4 spaces for nested bullet point lists.

### Actual behavior

```log
example-rumdl.md:7:1: [MD007] Expected 3 spaces for indent depth 1, found 4 [*]
example-rumdl.md:9:1: [MD007] Expected 3 spaces for indent depth 1, found 4 [*]
```

### Configuration

```toml
# ...
[MD007]
    # Unordered list indentation
    indent         = 4
    start-indented = false
# ...
```

```bash
$ uvx rumdl config get MD007.indent
MD007.indent = 4 [from project config]
```

### Operating system

Linux

**Comment by lucaspar** [comment]
It seems setting style in the config makes it behave the way I would expect it to. But I don't know the side-effects of it. Should this be the default? It's strange being asked for a 3-space indent when 4 is configured.

```toml
[MD007]
    # ...
    style = "fixed"
    # ...
```

**Comment by rvben** [comment]
Fixed in v0.1.44. MD007 now correctly accepts both text-aligned and fixed indent under ordered parents with `indent = 4` configuration. Thanks for reporting @lucaspar!

**Comment by lucaspar** [comment]
cool, thanks!
